### ISSUE TITLE
[Bug] AppShell: Issues after navigating away from the view hosting the SegmentControl

### ISSUE THREAD

**Issue by gavingreenhorn** [thomasgalliker/Plugin.SegmentedControl.Maui#20]
#### Description

Unexpected behavior when navigating away from the view hosting the SegmentControl and back to it (Android platform)

#### Steps to Reproduce

(in the reproduction project)
1.  Navigate to the first flyout item, a tabbed page is opened with two separate views, the initially opened hosting segmented control that works as expected. 
2. Navigate to another flyout item (Home), or another tab within the page, or switch the view within the same tab (first picker on this page)
3.  Upon returning to the view the segmented control breaks it layout positioning and doesn't change selected item.

#### Basic Information

- Version with issue: 1.3.7 ( .NET 8.0 )
- Last known good version: -

#### Screenshots

Works on initial navigation:
![image](https://github.com/user-attachments/assets/d74032e5-33a2-45f6-aa21-03d4da7deac0)

After returning from another page: selected item can't be changed (although segment is changing in UI), something with styles (de-selected segment retain white font in this case)
![image](https://github.com/user-attachments/assets/7b5b9baa-9a47-42e5-a5ac-0485578ff6ba)

After selecting another view within the same page then returning to this one, it also ignores layout positioning in grid (although retain it relative to another segmented control)
![image](https://github.com/user-attachments/assets/79b8b83c-6681-4615-9893-228ce87aafcc)

#### Reproduction Link
I've created a repository with minimal example here: https://github.com/gavingreenhorn/TestRepo/tree/master/SegmentedControlReproduce

**Comment by thomasgalliker** [comment]
I‘m sorry for not having looked into this yet. It‘s quite busy times right now.

**Comment by gavingreenhorn** [comment]
No worries at all, thank you for your work.

**Comment by thomasgalliker** [comment]
In the meantime, can you set the `AutoDisconnectHandler` property in SegmentedControl to `false` and check if you still have the error?

I assume this error has to do with the auto-disconnect feature, which should automatically call DisconnectHandler as soon the the SegmentedControl is no longer used on any page.

**Comment by gavingreenhorn** [comment]
It didn't help unfortunately, the control still breaks on changing active view.

**Comment by thomasgalliker** [comment]
It looks like MAUI does not maintain the `NavigationStack `of `Shell.Current` correctly. To be honest, I don't know the AppShell very well. Do you know if there is any chance to check if a certain page (e.g. BlankPage) is still part of the NavigationStack?

**Comment by thomasgalliker** [comment]
If you set `AutoDisconnectHandler="False"` on each instance of `SegmentedControl`, it seems to work:
```
 <s:SegmentedControl
     x:Name="SegControl1"
     AutoDisconnectHandler="False"
     ItemsSource="{Binding VowelVisualizationOptions}"
     SelectedItem="{Binding SelectedVowelVisualizationOption, Mode=TwoWay}"
     TextPropertyName="Thai"
     TintColor="Indigo"
     FontSize="22"
     Margin="2,0">
 </s:SegmentedControl>
 <s:SegmentedControl
     Grid.Row="1"
     x:Name="SegControl2"
     AutoDisconnectHandler="False"
     ItemsSource="{Binding SoundTypeOptions}"
     SelectedItem="{Binding SelectedSoundTypeOption, Mode=TwoWay}"
     TextPropertyName="Thai"
     TintColor="ForestGreen"
     FontSize="22"
     Margin="2,0">
 </s:SegmentedControl>
```

But, this is just a workaround! To give you a little insight: There is a class HandlerCleanUpHelper that helps to automatically call DisconnectHandler methods of SegmentedControlHandler. It's a tricky piece of code because it needs to detect when a SegmentedControl is no longer needed, means, when the parent/hosting page is unloaded AND no longer part of the navigation stack. As it looks like, I have no chance to check if a page is still on the navigation stack when you use MAUI's AppShell.

If you know more about the AppShell and how I can check if a page is currently part of the navigation stack, please let me know.

**Comment by gavingreenhorn** [comment]
Yes, this workaround works, thanks a lot!
Honestly, I don't know how shell navigation works behind the scenes.
One has no control over the first view opening by flyout menu navigation whatsoever, cannot replace its content view in the runtime, cannot reset it, the only workaround is instantiating the view from code behind, which feels wrong.
And at this point my codebase is like 60% workaround, I decidedly think that Shell App is fundamentally broken, and I'll probably ditch it altogether for either URI navigation, or some other framework altogether. Everything can probably be fixed or circumvented, but I'd much rather spend my efforts on the app I want to build using an instrument that doesn't actively fight me.
Had to fight bugs and unexpected/undocument behaviors every step of the way for a couple of month building a very simple app that I could build in WPF in a day or two.
Sorry for the rant. :D I think this issue can be closed with the resoultion "when the sun sets in the East and MAUI gets fixed".

**Comment by thomasgalliker** [comment]
@gavingreenhorn I did another small optimization but it should not break anything. I added your shell reproduction app to this repository, just in case we face more problems with shell apps in the future.

And yes, 100% agree with what your write abote. We put a lot of energy and time into fixing things in MAUI that used to work in Xamarin. I still believe, MAUI would be a great way to create cross-platform apps, but I also fear that the MAUI team is completely understaffed. I think they have a lot of very professional devs in the teams, but still, they cannot keep up with the flood of issues we see in the dotnet/maui repository.

Regarding AppShell: For production apps I'd highly recommend to use Prism.Maui. It's another layer of abstraction, yes, but they address a lot of issues that should actually be part of .NET MAUI. The whole navigation stuff is solved very nicely in Prism.
